### PR TITLE
docs: serializer utilities PTL-1699

### PR DESCRIPTION
**Pull request by Gareth-Spencer-Genesis**
ATTENTION! NEW DOCS STRUCTURE
FOR THE NEXT THREE WEEKS, ONLY BRANCH FROM new-docs-structure
OTHERWISE, WAIT UNTIL THE NEW STRUCTURE HAS BEEN RELEASED

  - Please check the [internal contributions guide](https://www.notion.so/genesisglobal/Contributing-new-documentation-75953fb245f246ff872789035451a0c4) for details
  
__________

Thank you for contributing to the documentation.

Do the changes you have made apply to both Current and Previous versions?
Yes

Have you done a trial build to check all new or changed links?
No

Is there anything else you would like us to know?
Yes

**This week's exciting advice from the style guide**

- Write your headings in sentence case:

  - A good example
    This is a correct heading.
  - A Bad Example
    This is not a correct heading.

### CODE REVIEW

File: docs/001_develop/03_client-capabilities/014_utility-methods/11_serializer-util.mdx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+## Examples
+
+### JSON Serialization/Deserialization
+
+The `JSONSerializer` DI Token enable the serialization and deserialization of complex objects, including BigInt values and custom object types, ensuring that JSON data is accurately represented in JavaScript.
+
+```typescript
+import { JSONSerializer } from '@genesislcap/foundation-utils';
+
```
Comment (mahamzafar990) [inline docs/001_develop/03_client-capabilities/014_utility-methods/11_serializer-util.mdx:33]: use GenesisElement here instead of FASTElement

**Review by mahamzafar990 — COMMENTED** [review]
looks good. Just update the code example as mentioned in comment

File: docs/001_develop/03_client-capabilities/014_utility-methods/11_serializer-util.mdx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+id: serializer-util
+keywords: [utils, utility, serializer, serialization, deserialization, complex, robust]
+tags:
+  - utils
+  - utility
+  - serializer
+  - serialization
+  - deserialization
+  - complex
+  - robust
```
Comment (matteematt) [inline docs/001_develop/03_client-capabilities/014_utility-methods/11_serializer-util.mdx:13]: Let's get some more keywords in here relating to each of the individual functions, especially to JSON etc. This is a useful utility that people ask about a lot

File: docs/001_develop/03_client-capabilities/014_utility-methods/11_serializer-util.mdx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+---
+
+This contains a comprehensive JSON serialization and deserialization solution, designed to support complex data structures, including those involving big integers and custom object types, ensuring robust and flexible handling of JSON data.
+
+## Key Features
+
+- **`customNumberParser` Function:** Enhances the parsing of numeric values, supporting safe integers, floating-point numbers, and BigInt values.
+- **`JSONSerializer` Interface:** Defines methods for serializing objects to JSON strings and deserializing JSON strings to JavaScript objects, with support for HTTP response objects and message event data.
+- **`DefaultJSONSerializer` Class:** Implements the `JSONSerializer` interface, providing default serialization and deserialization behaviors that can be customized via the `JSONSerializerConfig`.
+- **`JSONReplacer` and `JSONReviver` Functions:** Facilitate custom serialization and deserialization processes, allowing for the handling of special types, such as dates and NaN values, and the exclusion of properties with specific key prefixes.
```
Comment (matteematt) [inline docs/001_develop/03_client-capabilities/014_utility-methods/11_serializer-util.mdx:23]: Can we get an example of tis this please?

File: docs/001_develop/03_client-capabilities/014_utility-methods/11_serializer-util.mdx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+## Examples
+
+### JSON Serialization/Deserialization
+
+The `JSONSerializer` DI Token enable the serialization and deserialization of complex objects, including BigInt values and custom object types, ensuring that JSON data is accurately represented in JavaScript.
+
```
Comment (matteematt) [inline docs/001_develop/03_client-capabilities/014_utility-methods/11_serializer-util.mdx:30]: Can you add a tip here linking how to do the dependency injection handling in [angular](https://github.com/genesiscommunitysuccess/docs/blob/584d22aeabd7e1073fdcf05b7aa8c17150403f47/docs/001_develop/03_client-capabilities/022_framework-integration/002_integration-angular.mdx?plain=1#L311) and [react](https://github.com/genesiscommunitysuccess/docs/blob/584d22aeabd7e1073fdcf05b7aa8c17150403f47/docs/001_develop/03_client-capabilities/022_framework-integration/003_integration-react.mdx?plain=1#L275)

**Review by matteematt — APPROVED** [review]
Discussed on slack and we're both a bit confused about the features of the JSON serializer. Getting this merged for now and then Gareth will carry on looking into it after

**Pull request merged**